src/public.js

```diff
@@ -4,17 +4,18 @@
  */
 
 import Vue from 'vue'
+import { getCurrentUser } from '@nextcloud/auth'
 import UserApp from './UserApp.vue'
 
 Vue.prototype.t = t
 Vue.prototype.n = n
 Vue.prototype.OC = OC
 Vue.prototype.OCA = OCA
 
-const hasToken = (document.getElementById('sharingToken') !== null)
+const isPublicPage = getCurrentUser() === null
```

Comment:
```suggestion
const isGuest = getCurrentUser() === null
```
or
```suggestion
const isNotLoggedIn = getCurrentUser() === null
```

as e.g. with talk it's not a "public page" per se